src/main/java/com/bp3x/raidbot/commands/lfg/LFGReactionListener.java

```diff
@@ -0,0 +1,81 @@
+package com.bp3x.raidbot.commands.lfg;
+
+import com.bp3x.raidbot.commands.lfg.util.Event;
+import com.bp3x.raidbot.commands.lfg.util.LFGEmbedBuilder;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.message.react.GenericMessageReactionEvent;
+import net.dv8tion.jda.api.events.message.react.MessageReactionAddEvent;
+import net.dv8tion.jda.api.events.message.react.MessageReactionRemoveEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+
+public class LFGReactionListener extends ListenerAdapter {
+
+    Logger log = LoggerFactory.getLogger(LFGReactionListener.class);
+
+    @Override
+    public void onMessageReactionAdd(MessageReactionAddEvent addEvent) {
+       handleReactionEvent(addEvent);
+    }
+
+    @Override
+    public void onMessageReactionRemove(MessageReactionRemoveEvent removeEvent) {
+        handleReactionEvent(removeEvent);
+    }
+
+    /**
+     * Method to handle either a MessageReactionAddEvent or MessageReactionRemoveEvent to update the embed then remove the reaction.
+     * @param reactionEvent the MessageReaction event needing to be handled
+     */
+    private void handleReactionEvent(GenericMessageReactionEvent reactionEvent) {
+        String reactionEmoji = reactionEvent.getReactionEmote().getName();
+
+        // ignore bot reactions
+        if (reactionEvent.getUser().isBot() || !LFGConstants.EMOJI_LIST.contains(reactionEmoji)) return;
+
+        else {
+            Message message = reactionEvent.getChannel().retrieveMessageById(reactionEvent.getMessageId()).complete();
```

Comment:
Did a little extended reading and it looks like [`.submit()`](https://github.com/DV8FromTheWorld/JDA/wiki/7%29-Using-RestAction#using-submit) might be what we want here. We could try it and see if it does what we need so that we don't need to use `.complete()` that blocks the thread.

Reply:
If so, we should look into replacing all uses of `.complete()` with `.submit()`! That's fine to be a separate commit though.